File: services/delivery/src/main/java/com/example/iter/delivery/domain/repository/ShippingRepository.java
package com.example.iter.delivery.domain.repository;

import com.example.iter.delivery.domain.entity.Shipping;
import com.example.iter.delivery.domain.entity.ShippingType;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface ShippingRepository extends JpaRepository<Shipping, Long> {
    List<Shipping> findByRentalId(Long rentalId);

    Optional<Shipping> findByRentalIdAndType(Long rentalId, ShippingType type);
}

File: services/delivery/src/main/kotlin/iter/delivery/domain/repository/ShippingRepository.kt
package iter.delivery.domain.repository

import iter.delivery.domain.entity.Shipping
import iter.delivery.domain.entity.ShippingType
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface ShippingRepository : JpaRepository<Shipping, Long> {
    fun findByRentalId(rentalId: Long): List<Shipping>

    fun findByRentalIdAndType(rentalId: Long, type: ShippingType): Optional<Shipping>
}
